src/UglyToad.PdfPig/Graphics/Colors/ColorSpaceDetails.cs

```diff
@@ -1239,6 +1239,12 @@ public override IColor GetColor(params double[] values)
 
             // TODO - use ICC profile
 
+            for (int c = 0; c < values.Length; c++)
+            {
+                int i = 2 * c;
+                values[c] = PdfFunction.ClipToRange(values[c], Range[i], Range[i + 1]);

```

Comment:
Out of interest what is happening here?

Reply:
This is tested in `RepresentativeWindEnergyDeals()` in the `GenerateLetterGlyphImages` (doument is GHOSTSCRIPT-698168-0.pdf)

In this particular test, the color space is ICC, but we don't support it. So we use the Alternate CC.


> Range
> (Optional) An array of 2 × N numbers [𝑚𝑖𝑛0 𝑚𝑎𝑥0 𝑚𝑖𝑛1 𝑚𝑎𝑥1…] that shall specify the minimum and maximum valid values of the corresponding colour components. These values shall match the information in the ICC profile. Default value: [0.0 1.0 0.0 1.0…].

Reply:
I think this is how they do in PdfBox, where they clamp values too.

https://github.com/apache/pdfbox/blob/33ebd39096a650df157a846d9e8c631a1e3631ab/pdfbox/src/main/java/org/apache/pdfbox/pdmodel/graphics/color/PDICCBased.java#L291C13-L291C75

Again, happy for you to tell me if you understand that differently